### PR TITLE
update guide of SYCL backend

### PR DESCRIPTION
**Pull request by NeoZhangJianyu**
update guide:
make tool installation
memory choose
gguf model link
rm todo for windows build

### CODE REVIEW

**Comment by NeoZhangJianyu** [comment]
@airMeng , @luoyu-intel , @abhilash1910, @ggerganov Invite you to review if you are idle.

**Review by airMeng — APPROVED** [review]

File: README-sycl.md [hunk 1]
```diff
@@ -50,6 +52,17 @@ For Intel CPU, recommend to use llama.cpp for X86 (Intel MKL building).
 |Intel built-in Arc GPU| Support| built-in Arc GPU in Meteor Lake|
 |Intel iGPU| Support| iGPU in i5-1250P, i7-1165G7|
 
+Note: If the EUs (Execution Unit) in iGPU is less than 80, the inference speed will too slow to use.
```
Comment (abhilash1910) [inline README-sycl.md:55]:
```suggestion
Note: If the EUs (Execution Unit) in iGPU is less than 80, the inference speed will be too slow to use.
```

File: README-sycl.md [hunk 2]
```diff
@@ -50,6 +52,17 @@ For Intel CPU, recommend to use llama.cpp for X86 (Intel MKL building).
...
+
+### Memory
+
+The memory is a limitation to run LLM on GPUs.
+
+When run llama.cpp, there is print log to show the applied memory on GPU. You could know how many memory to be used in your case. Like `llm_load_tensors:            buffer size =  3577.56 MiB`.
```
Comment (abhilash1910) [inline README-sycl.md:61]:
```suggestion
When run llama.cpp, there is print log to show the applied memory on GPU. You could know how much memory to be used in your case. Like `llm_load_tensors:            buffer size =  3577.56 MiB`.
```

File: README-sycl.md [hunk 3]
```diff
@@ -105,7 +118,7 @@ source /opt/intel/oneapi/setvars.sh
 sycl-ls
 ```
 
-There should be one or more level-zero devices. Like **[ext_oneapi_level_zero:gpu:0]**.
+There should be one or more level-zero devices. Please confirm that at least one GPU be present, like **[ext_oneapi_level_zero:gpu:0]**.
```
Comment (abhilash1910) [inline README-sycl.md:121]:
```suggestion
There should be one or more level-zero devices. Please confirm that at least one GPU is present, like **[ext_oneapi_level_zero:gpu:0]**.
```

File: README-sycl.md [hunk 4]
```diff
@@ -419,8 +448,25 @@ Using device **0** (Intel(R) Arc(TM) A770 Graphics) as main device
...
+- Meet compile error.
+
+  Remove folder **build** and try again.
+
+- I can **not** see **[ext_oneapi_level_zero:gpu:0]** afer install GPU driver in Linux.
 
-- Support to build in Windows.
+  Please run **sudo sycl-ls**.
+
+  If see it in result, please add video/render group to your ID:
```
Comment (abhilash1910) [inline README-sycl.md:459]:
```suggestion
  If you see it in result, please add video/render group to your ID:
```

File: README-sycl.md [hunk 5]
```diff
@@ -419,8 +448,25 @@ Using device **0** (Intel(R) Arc(TM) A770 Graphics) as main device
...
+
+  ```
+  sudo usermod -aG render username
+  sudo usermod -aG video username
+  ```
+
+  Then **relogin**.
+
+  If no see it, please check the installation GPU steps again.
```
Comment (abhilash1910) [inline README-sycl.md:468]:
```suggestion
  If you do not see it, please check the installation GPU steps again.
```

File: README-sycl.md [hunk 6]
```diff
@@ -260,15 +277,21 @@ Output (example):
...
 ```
 
 3. Install cmake & make
 
-a. Download & install cmake for windows: https://cmake.org/download/
+a. Download & install cmake for Windows: https://cmake.org/download/
+
+b. Download & install make for Windows provided by mingw-w64
+
+- Download binary package for Windows in https://github.com/niXman/mingw-builds-binaries/releases.
```
Comment (luoyu-intel) [inline README-sycl.md:292]: VS build is more common for Windows users. Just a thought.

File: README-sycl.md (hunk 1 above)
Comment (NeoZhangJianyu) [reply README-sycl.md:55]: Yes, update

File: README-sycl.md (hunk 2 above)
Comment (NeoZhangJianyu) [reply README-sycl.md:61]: yes

File: README-sycl.md (hunk 3 above)
Comment (NeoZhangJianyu) [reply README-sycl.md:121]: yes

File: README-sycl.md [hunk 7]
```diff
@@ -252,23 +273,29 @@ In oneAPI command line:
 sycl-ls
 ```
 
-There should be one or more level-zero devices. Like **[ext_oneapi_level_zero:gpu:0]**.
+There should be one or more level-zero devices. Please confirm that at least one GPU be present, like **[ext_oneapi_level_zero:gpu:0]**.
```
[the remark this answers is not in the document]
Comment (NeoZhangJianyu) [reply README-sycl.md:276]: yes

File: README-sycl.md (hunk 5 above)
Comment (NeoZhangJianyu) [reply README-sycl.md:468]: yes

File: README-sycl.md (hunk 4 above)
Comment (NeoZhangJianyu) [reply README-sycl.md:459]: yes

File: README-sycl.md (hunk 6 above)
Comment (NeoZhangJianyu) [reply README-sycl.md:292]: Yes. But I meet this issue in a new PC.
I think it's better record here.

**Review by abhilash1910 — APPROVED** [review]

File: README-sycl.md (hunk 6 above)
Comment (cebtenzzre) [reply README-sycl.md:292]: We recommend w64devkit for MinGW in the main README, I don't know why we would recommend yet another MinGW distribution here.

Comment (NeoZhangJianyu) [reply README-sycl.md:292]: Here use the make.exe in MinGW only.
w64devkit is docker env.
In the chapter **Windows**, we provide the native build method. So not use w64devkit.
In the chapter **Docker**, there is docker method.

I will find another method to install make.exe without MinGW.

Thank you!

Comment (NeoZhangJianyu) [reply README-sycl.md:292]: @cebtenzzre Update as your suggestion: https://github.com/ggerganov/llama.cpp/pull/5297

**Pull request merged**